app-libs/stf/poc/Cargo.toml

```diff
@@ -6,6 +6,16 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-serde = { version = "1.0.147", features=['derive'] }
-serde_json = "1.0.87"
-match-test = { path = "match-test" }
\ No newline at end of file
+serde = { version = "1.0.147", default-features=false, features=['derive'] }
+serde_json = { version = "1.0.87", default-features=false }
+match-test = { path = "match-test", default-features=false }
+log = { version = "*", default-features=false}
+
+[features]
+default = ["std"]
+std = [
+    "log/std",
+    "match-test/std",
+    "serde/std",
+    "serde_json/std",
+]
```

Comment:
You can decide to skip this if you want your crate to be pure no_std, this is your decision. You might want to keep this:

* Maybe some features can't be implemented in no_std, but you want to have parts of it available in no_std.
* Many things are more efficiently implemented in std, then you might want to add this so your dependants have access to the more efficient code.